packages/strapi-plugin-users-permissions/admin/src/components/Permissions/index.js

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback, useMemo } from 'react';
+import { Padded } from '@buffetjs/core';
+
+import { useUsersPermissions } from '../../contexts/UsersPermissionsContext';
+import ListWrapper from './ListWrapper';
+import PermissionRow from './PermissionRow';
+
+const Permissions = () => {
+  const { permissions, pluginName, onSetPluginName } = useUsersPermissions();
+
+  const handleOpenPlugin = useCallback(
+    name => {
+      onSetPluginName(name === pluginName ? null : name);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [pluginName]
+  );
+
+  const formattedPermissions = useMemo(() => {
```

Comment:
It would be great to have some small tests so we can more easily understand the data your are manipulating. What do you think?

You can extract your code into a small util so you don't have a lot of changes to make.